### PR TITLE
V5.0.0

### PR DESCRIPTION
**Pull request by davidteather**
**IMPORTANT CHANGES**
* Please read the upgrading from V4 to V5 section on the readme
    * Higher level approach using objects
    * Auto-parsing of attributes and responses back into objects
    * Should be easier to work with (especially for new-comers not as familiar with this package)
* dropped support for selenium (sorry but it's been too annoying to main multiple browsers)

TODO
* None

DONE
* New YouTube video tutorial for v5
* refactor kwargs stuff so it's not as bad as it is right now
* make sure everything is typed and valid
* write a section in the api modules about what different keys mean and their types
* split hashtag, video, sound, and user into their own classes
    * creates a higher level feel of the package
* switch to generators for all the methods
* update tests
* improve logging
* Write an upgrading to v5 section in the readme
* Add more doc strings to the methods everywhere
* update readme to reflect new changes
* update issue templates
* change process_kwargs
* try to minimize amount of instance variables on TikTokApi for cleanliness or at least use private instance variables

V5 Example Usage
```py
from TikTokApi import TikTokApi

verify_fp = "verify_xxx"
api = TikTokApi(custom_verify_fp=verify_fp)

tag = api.hashtag(name="funny")
print(tag.info())

for video in tag.videos():
    print(video.id)
    for u_vid in video.author.videos():
        print(u_vid.id)
```

### CODE REVIEW

**Comment by estatistics** [comment]
how to download a user's videos; can you provide an example?

**Review by 3garbilx — COMMENTED** [review]
[README.md](https://github.com/davidteather/TikTok-Api/files/10949942/README.md)

File: tests/test_user.py [hunk 1]
```diff
@@ -1,52 +1,41 @@
 from TikTokApi import TikTokApi
```
Comment (3garbilx) [inline tests/test_user.py:1]: 1

File: tests/test_user.py [hunk 2]
```diff
@@ -1,52 +1,41 @@
...
 import os
```
Comment (3garbilx) [inline tests/test_user.py:2]: 2

**Review by 3garbilx — COMMENTED** [review]
72

**Review by 3garbilx — COMMENTED** [review]
gh pr checkout 803

**Pull request merged**